mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -465,33 +497,64 @@ private[ml] trait GBTParams extends TreeEnsembleParams with HasMaxIter with HasS
 }
 
 private[ml] object GBTClassifierParams {
-  // The losses below should be lowercase.
-  /** Accessor for supported loss settings: logistic */
-  final val supportedLossTypes: Array[String] = Array("logistic").map(_.toLowerCase)
+  // The values below should be lowercase.
+  /** Accessor for supported loss settings: logistic, bernoulli */
+  final val supportedLossTypes: Array[String] = Array("logistic", "bernoulli")
+  /** Accessor for support entropy settings: loss-based or variance */
+  final val supportedImpurities: Array[String] = Array("loss-based", "variance")
+  final def getLossBasedImpurity(loss: String): OldImpurity = loss match {
+    case "logistic" | "bernoulli" => ApproxBernoulliImpurity
+    case _ => throw new RuntimeException(
+      s"GBTClassifier does not have loss-based impurity for loss ${loss}")
+  }
 }
 
 private[ml] trait GBTClassifierParams extends GBTParams with TreeClassifierParams {
 
   /**
+   * Criterion used for information gain calculation (case-insensitive).
+   * Also used for terminal leaf value prediction.
+   * Supported: "loss-based" (default) and "variance"
+   *
+   * @group param
+   */
+  override val impurity: Param[String] = new Param[String](this, "impurity", "Criterion used for" +
+    " information gain calculation (case-insensitive). Supported options:" +
+    s" ${GBTClassifierParams.supportedImpurities.mkString(", ")}",
+    (value: String) => GBTClassifierParams.supportedImpurities.contains(value.toLowerCase))
+
+  /** (private[ml]) convert new impurity to old impurity */
```

Comment:
No need for "(private[ml])" note anymore; this was from previous releases when Java doc gen did not handle package private correctly.